src/tree.rs

```diff
@@ -215,24 +214,31 @@ impl std::hash::Hash for OrchardIncrementalTreeDigest {
     }
 }
 
+impl ConditionallySelectable for OrchardIncrementalTreeDigest {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        OrchardIncrementalTreeDigest(pallas::Base::conditional_select(&a.0, &b.0, choice))
+    }
+}
+
 impl Hashable for OrchardIncrementalTreeDigest {
     fn empty_leaf() -> Self {
-        OrchardIncrementalTreeDigest(CtOption::new(*UNCOMMITTED_ORCHARD, 1.into()))
+        OrchardIncrementalTreeDigest(*UNCOMMITTED_ORCHARD)
     }
 
-    fn combine(altitude: Altitude, left_opt: &Self, right_opt: &Self) -> Self {
-        OrchardIncrementalTreeDigest(left_opt.0.and_then(|left| {
-            right_opt
-                .0
-                .and_then(|right| hash_with_l(altitude.into(), Pair { left, right }))
-        }))
+    fn combine(altitude: Altitude, left: &Self, right: &Self) -> Self {
+        hash_with_l(
```

Comment:
Should `hash_with_l` internally map the bottom case to `pallas::Base::zero()` instead?

Reply:
I think that `hash_with_l` corresponds precisely to the specified:

```
where hash = SinsemillaHash(“ z.cash:Orchard-MerkleCRH ”, l⋆ || left⋆ || right⋆)
```
so I think it's worth keeping as it is.

Reply:
I agree, since `hash_with_l` is basically MerkleCRH.

Reply:
I was replying to @therealyingtong, so my comment is confusing. I'm fine with `hash_with_l` existing, but in that case, add a doc comment that this function is implementing `MerkleCRH` (see documentation examples in `src/bundle.rs`).

Reply:
MerkleCRH includes the mapping to zero; `hash_with_l` does not.

Reply:
Reading the full PR again, I now agree with @nuttycom and @daira.